boards/xtensa/esp32s3_luatos_core/esp32s3_luatos_core.dts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2023 YuLong Yao <feilongphone@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32s3/esp32s3_mini_n8.dtsi>
+#include "esp32s3_luatos_core-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "esp32s3_luatos_core";
+	compatible = "espressif,esp32s3";
+
+	aliases {
+		led0 = &led0;
+		led1 = &led1;
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+	};
+
+	aliases {
+		uart-0 = &uart0;
+		sw0 = &button0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led0 {
+			gpios = <&gpio0 10 GPIO_ACTIVE_HIGH>;
+			label = "LEDA";
+		};
+		led1: led1 {
+			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+			label = "LEDB";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpio0 0 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "BOOT Button";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&cpu1 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&usb_serial {
+	status = "okay";
```

Comment:
Is it necessary to enable usb_serial by default? There is CH343 driver on-board.

Reply:
Reference: https://wiki.luatos.com/chips/esp32s3/hardware.html

There is a switch (S2) on the development board to control the USB switch (S1),  
and the control PC is directly connected to ESP32 or connected through ch343.  

I don't understand the different of CH343 on the programming and debugging functions,   
but I think enable usb_serial can make it work in either mode.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10269596/cbdff405-6f69-45eb-8e6b-65957977285d)

Reply:
I realized I just misread that this is a usb serial port and not a debug port.

There is a switch (S2) on the development board to control the USB switch (S1),  
and the control PC is directly connected to ESP32 or connected through ch343.  

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10269596/cbdff405-6f69-45eb-8e6b-65957977285d)

So when the usb switch enables CH343 you are right, we don't need to use usb_serial.
But when the switch directly disables CH343, usb_serial needs to be turned on.

Reply:
Perhaps it would be better to provide two dts files corresponding to whether CH343 is enabled or not?

``` kconfig
# Kconfig.board

# ESP32S3 Core board configuration
config BOARD_ESP32S3_LUATOS_CORE_NOCH343
	bool "ESP32S3 LuatOS Core Board with ch343 disabled"
	depends on SOC_SERIES_ESP32S3

config BOARD_ESP32S3_LUATOS_CORE_CH343
	bool "ESP32S3 LuatOS Core Board with ch343 enabled"
	depends on SOC_SERIES_ESP32S3

```

Reply:
I commented on this because, on (both) of my `esp32s3_luatos` boards, the `S2` switch default position is `CH343`. So I assume if CDC is needed, it would be enabled at the application overlay.